test/apiv2/40-pods.at

```diff
@@ -35,12 +45,72 @@ else
 fi
 t POST  libpod/pods/foo/unpause '' 200
 t POST "libpod/pods/foo/unpause (2nd unpause in a row)" '' 200
+t POST "libpod/pods/fakename/unpause" '' 404\
+  .cause="no such pod" \
+  .message="no pod with name or ID fakename found: no such pod"
+
+
+t POST libpod/pods/foo/stop '' 200 \
+  .Errs=null \
+  .Id=$pod_id
+
 t POST "libpod/pods/foo/stop (pod is already stopped)"  '' 304
+t POST "libpod/pods/fakename/stop" '' 404\
+  .cause="no such pod" \
+  .message="no pod with name or ID fakename found: no such pod"
+
 t POST  libpod/pods/foo/restart '' 200 \
   .Errs=null \
   .Id=$pod_id
 
 t POST  "libpod/pods/bar/restart (restart on nonexistent pod)" '' 404
+t POST libpod/pods/create name=bar 201 .Id~[0-9a-f]\\{64\\}
+pod_bar_id=$(jq -r .Id <<<"$output")
+
+t POST  libpod/pods/bar/restart '' 200 \
+  .Errs=null \
+  .Id=$pod_bar_id
+
+t GET  libpod/pods/bar/json        200 \
+  .State=Running
+
+t POST  libpod/pods/bar/restart '' 200 \
+  .Errs=null \
+  .Id=$pod_bar_id
+
+t POST "libpod/pods/bar/stop?t=invalid" '' 400 \
+  .cause="schema: error converting value for \"t\"" \
+  .message~"Failed to parse parameters for"
+
+podman run -d --pod bar busybox sleep 999
+
+t POST libpod/pods/bar/stop?t=1 '' 200 \
+  .Errs=null \
+  .Id=$pod_bar_id
+
+t POST libpod/pods/bar/start '' 200
+
+t GET libpod/pods/stats?all=true 200
+is `awk -F"{" '{print NF-1}' <<< "$output"` 3 "GET libpod/pods/stats?all=true"
```

Comment:
This and line 97 are difficult to understand; they're also fragile, because it's possible that a future version of `stats` could return nested objects. Did you know you can use `jq` itself to get an array length? I would encourage you to replace this (and line 97) with:
```bash
is $(jq '. | length' <<<"$output") 3 "stats?all=true: number of records found"
```
Note also the change in the message: there is no need to duplicate the entire `GET etc etc', because that is already shown in the previous line. This form removes unnecessary text, and also gives information on what is actually being tested (number of matches).

Reply:
Thanks @edsantiago . Will update it.